test/presenters/taxon_presenter_test.rb

```diff
@@ -49,4 +52,150 @@
       TaxonPresenter.new(taxon).accordion_content
     end
   end
+
+  describe 'options' do
+    let(:content_hash) { funding_and_finance_for_students_taxon }
+    let(:content_item) { ContentItem.new(content_hash) }
+    let(:taxon) { Taxon.new(ContentItem.new(content_hash)) }
+    let(:child_taxon) { taxon.child_taxons.first }
+    let(:taxon_presenter) { TaxonPresenter.new(taxon) }
+
+    describe '#options_for_leaf_content' do
+      before :each do
+        @search_results = generate_search_results(15)
+        stub_content_for_taxon([taxon.content_id], @search_results)
+      end
+      describe 'root options' do
+        subject { taxon_presenter.options_for_leaf_content(index: 10) }
+        it 'contains the track-click module' do
+          assert_equal('track-click', subject[:module])
+        end
+        it 'contains the navLeafLinkClicked track_category' do
+          assert_equal('navLeafLinkClicked', subject[:track_category])
+        end
+        it 'contains track-action equal to the index + 1' do
+          assert_equal('11', subject[:track_action])
+        end
+        it 'contains track_label equal to the base path of the content' do
+          assert_equal(@search_results[10]['link'], subject[:track_label])
+        end
+      end
+      describe 'dimensions' do
+        subject { taxon_presenter.options_for_leaf_content(index: 10)[:track_options] }
+        describe 'dimension 28 - number of tagged content items' do
+          it 'contains the number of tagged content items' do
+            assert_equal('15', subject[:dimension28])
```

Comment:
this could use some [custom matchers](https://relishapp.com/rspec/rspec-expectations/v/3-3/docs/custom-matchers/define-matcher) to really make it readable